src/compiler/scala/tools/nsc/settings/StandardScalaSettings.scala

```diff
@@ -52,7 +51,26 @@ trait StandardScalaSettings { _: MutableSettings =>
   val nowarn =         BooleanSetting ("-nowarn", "Generate no warnings.") withAbbreviation "--no-warnings" withPostSetHook { s => if (s) maxwarns.value = 0 }
   val optimise:        BooleanSetting // depends on post hook which mutates other settings
   val print =          BooleanSetting ("-print", "Print program with Scala-specific features removed.") withAbbreviation "--print"
-  val target =         ChoiceSetting  ("-target", "target", "Target platform for object files.", AllTargetVersions, "8") withPreSetHook normalizeTarget _ withAbbreviation "--target"
+  val release =
+    ChoiceSetting("-release", "release", "Compile for a version of the Java API and target class file.", AllTargetVersions, normalizeTarget(javaSpecVersion))
+      .withPostSetHook { setting =>
+        val current = setting.value.toInt
+        if (!isJavaAtLeast("9") && current > 8) errorFn.apply("-release is only supported on JVM 9 and higher")
+        if (target.valueSetByUser.map(_.toInt > current).getOrElse(false)) errorFn("-release cannot be less than -target")
```

Comment:
I think if `target` and `release` are both set then they should match exactly. If that's not  case, however, we should probably raise a warning rather than an error for compatibility with earlier compiler versions

Reply:
I gave this a bit of thought. They are adding features that are target-dependent, see the fun work by harpocrates, but not API-dependent. But probably there is no use case for release > target. I don't know how to warn from settings, but maybe it's OK to silently bump the target.